### PR TITLE
board: arm: add ubx_bmd340eval_nrf52840

### PR DESCRIPTION
**Pull request by ghost**
Add support for u-blox BMD-340-EVAL EVK
Tested with blinky, button, and peripheral_hr

Signed-off-by: Bob Recny <bob.recny@u-blox.com>

### CODE REVIEW

**Comment by brec-u-blox** [comment]
Looking for best practices guidance.
The test is failing on the ADC driver. This board has the same ADC capabilities as the Nordic `nRF52840dk_nrf52840`. 

Should I just add `CONFIG_BOARD_BMD340EVAL_NRF52840` to the list in this line:
`#elif defined(CONFIG_BOARD_NRF21540DK_NRF52840) || \`

**Review by ioannisg — APPROVED** [review]

**Comment by brec-u-blox** [comment]
@ioannisg I see the "Release Notes Required" label. Still a bit new to Zephyr. Where should these be added?

File: boards/arm/ubx_bmd340eval_nrf52840/Kconfig [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# BMD -340-EVAL board configuration
+
+# Copyright (c) 2021 u-blox AG
+# SPDX -License-Identifier: Apache-2.0
```
Comment (MaureenHelm) [inline boards/arm/ubx_bmd340eval_nrf52840/Kconfig:4]: There's an extra space between "SPDX" and "-License-Identifier" that needs to be removed

File: tests/drivers/adc/adc_api/src/test_adc.c [hunk 2]
```diff
@@ -1,4 +1,4 @@
-/*
+CONFIG_BOARD_BMD340EVAL_NRF52840/*
```
Comment (MaureenHelm) [inline tests/drivers/adc/adc_api/src/test_adc.c]: This doesn't look right. Rebase mistake?

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Thanks for the contribution!

Please squash the fixup changes into their associated original commits and force push. You also need to rebase and resolve a merge conflict.

Also, can you reduce the size of the doc images?

File: tests/drivers/adc/adc_api/src/test_adc.c (hunk 2 above)
Comment (brec-u-blox) [reply tests/drivers/adc/adc_api/src/test_adc.c]: No, copy/paste mistake. That should be fixed now.

File: boards/arm/ubx_bmd340eval_nrf52840/Kconfig (hunk 1 above)
Comment (brec-u-blox) [reply boards/arm/ubx_bmd340eval_nrf52840/Kconfig]: Fixed, also removed space in BMD-340-EVAL in line 1

**Comment by brec-u-blox** [comment]
I got something fouled up on the last rebase. Not sure how to undo the last force push. Best to abandon this PR and create a new one?

**Comment by dcpleung** [comment]
> I got something fouled up on the last rebase. Not sure how to undo the last force push. Best to abandon this PR and create a new one?

You can checkout the old commit ID https://github.com/zephyrproject-rtos/zephyr/commit/b718f041ed226eb87b2c074a93b2df82ce86ccd5 on your local branch and force push again.

**Comment by brec-u-blox** [comment]
No joy. I keep getting a message that everything is up to date after I checkout the previous version, make my edits again and force push. Somehow a bunch of other files not related to my PR got committed and I can seem to undo that.

**Comment by joerchan** [comment]
> No joy. I keep getting a message that everything is up to date after I checkout the previous ID, make my edits again and force push. Somehow a bunch of other files not related to my PR got committed and I can seem to undo that.

@brec-u-blox Looks like you managed to get some commits in between yours, I suggest you do the following, assuming `upstream` is the remote of zephyrproject-rtos/zephyr.

Update upstream master and start interactive rebase:
```
git fetch upstream master
git rebase -i 6478735493
```
You should then remove all the commits that came in between, it should end up like this:
```
pick 43f10e2509 board: arm: add ubx_bmd340eval_nrf52840
pick 1b3b50d4cf board: arm: updating bmd340 submission for adc tests
pick 8da8add973 board: arm: Updated adc test to include new board
pick a4d78060d5 board: arm: add BMD-340-EVAL
```

Then rebase and force push to your branch:
```
git rebase upstream/master
git push -f
```

**Comment by brec-u-blox** [comment]
@joerchan Thanks! that did the trick.

File: boards/arm/ubx_bmd340eval_nrf52840/Kconfig (hunk 1 above)
Comment (MaureenHelm) [reply boards/arm/ubx_bmd340eval_nrf52840/Kconfig]: It's fixed, but in the wrong commit. :( It's currently in `board: arm: updating bmd340 submission for adc tests` but should be in `board: arm: add ubx_bmd340eval_nrf52840`

File: boards/arm/ubx_bmd340eval_nrf52840/Kconfig [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+
+config BOARD_ENABLE_DCDC
+bool "Enable DCDC mode"
+select SOC_DCDC_NRF52X
+default y
+depends on BOARD_UBX_BMD340EVAL_NRF52840
```
Comment (MaureenHelm) [inline boards/arm/ubx_bmd340eval_nrf52840/Kconfig:10]: Indent by one tab

File: boards/arm/ubx_bmd340eval_nrf52840/ubx_bmd340eval_nrf52840.dts [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+	current-speed = <115200>;
+	rx-pin = <33>;
+	tx-pin = <34>;
+};
+
+arduino_i2c: &i2c0 {
+	compatible = "nordic,nrf-twi";
+	status = "okay";
+	sda-pin = <26>;
+	scl-pin = <27>;
```
Comment (MaureenHelm) [inline boards/arm/ubx_bmd340eval_nrf52840/ubx_bmd340eval_nrf52840.dts:168]: I2C pins need to be updated following #31965

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

**Comment by brec-u-blox** [comment]
I've got my local repositories all wound up into a knot. Restarting. Will apply all requested changes and issue a new pull request.

**Pull request closed without merging**